pandas/tests/io/parser/usecols.py

```diff
@@ -492,11 +492,11 @@ def test_raise_on_usecols_names_mismatch(self):
         tm.assert_frame_equal(df, expected)
 
         usecols = ['a', 'b', 'c', 'f']
-        with tm.assert_raises_regex(ValueError, msg):
+        with tm.assert_raises_regex(ValueError, msg.format(missing=['f'])):
```

Comment:
So we're repeating the same format for every scenario here, which seems silly, but if we were to extend this in the future it'd be useful to have this around (although trivial to implement if not) - would you rather me just have a string to check against and not bother formatting?

Reply:
Also, the error I'm getting during testing is super confusing:
```
E           AssertionError: "Usecols do not match columns, columns expected but not found: ['f']" does not match "Usecols do not match columns, columns expected but not found: ['f']"
```

Am I being silly? Is it to do with the regex matching?

Reply:
* Let's see if we do the string formatting.  We need to check that it raises on the right columns.
* Yes, it's regex matching.  So you'll need to add forward-slashes to the brackets.

Reply:
Sorry on the error regex matching, that's me being silly, thanks!